### PR TITLE
Feature/state size limits

### PR DESCRIPTION
**Pull request by liayoo**
- Use `MAX_STATE_TREE_SIZE_PER_BYTE` and `STATE_TREE_BYTES_LIMIT` to get `TREE_SIZE_BUDGET`
- Add tree size limits for service & apps

### CODE REVIEW

File: db/index.js [hunk 1]
```diff
@@ -952,12 +955,16 @@ class DB {
...
+  static updateStateGasAmount(tx, result, allStateUsageBefore, allStateUsageAfter, stateUsagePerAppBefore, stateUsagePerAppAfter) {
     const LOG_HEADER = 'updateStateGasAmounts';
+    const serviceTreeBytesDelta =
+        _.get(allStateUsageAfter, `service.${StateInfoProperties.TREE_BYTES}`, 0) -
+        _.get(allStateUsageBefore, `service.${StateInfoProperties.TREE_BYTES}`, 0);
     const stateGasAmount = {
-      service: Math.max(treeBytesAfter.service - treeBytesBefore.service, 0) * STATE_GAS_CONSTANT,
-      app: Object.keys(treeBytesPerAppAfter).reduce((acc, appName) => {
-          const delta = treeBytesPerAppAfter[appName] - treeBytesPerAppBefore[appName];
+      service: Math.max(serviceTreeBytesDelta, 0) * STATE_GAS_CONSTANT,
```
Comment (platfowner) [inline db/index.js:964]: minor: what about _CONSTANT -> _COEFFICIENT?

File: db/index.js [hunk 2]
```diff
@@ -952,12 +955,16 @@ class DB {
...
+      app: Object.keys(stateUsagePerAppAfter).reduce((acc, appName) => {
```
Comment (platfowner) [inline db/index.js:965]: For safety, could you make sure that this prop is not set when empty?

**Review by platfowner — APPROVED** [review]
LGTM with some minor comments.
Thanks!

File: db/index.js (hunk 1 above)
Comment (liayoo) [reply db/index.js]: Done! Thanks!

File: db/index.js (hunk 2 above)
Comment (liayoo) [reply db/index.js]: Thanks for the suggestion! Changed the deleting logic to not setting if the value is empty.

**Comment by liayoo** [comment]
Thanks for the quick review! 🙌 Now merging~

**Pull request merged**